docs/internals/optimizer.rst

```diff
@@ -528,6 +528,7 @@ The above would be transformed into
 
 .. code-block:: yul
 
+    // This will not compile (y and x are undefined)
```

Comment:
Can't we fix the example instead?

Reply:
I wasn't sure how much changes I can do to the examples without making them too convoluted. I could simply define x and y here yes. For the ones below involving `E` it was not as clear

Reply:
Just replace `x` and y` by constants.